### PR TITLE
ss-16: Add Double Rande Slider

### PR DESCRIPTION
**Pull request by MstyslavDmytryk**
slider

### CODE REVIEW

File: frontend/src/bundles/common/components/range-slider/range-slider.module.scss
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/range-slider/range-slider.module.scss:1]: styles.module.scss

File: frontend/src/bundles/common/components/range-slider/range-slider.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+import styles from './range-slider.module.scss';
+
+interface FinanceOperation {
+    amount: number;
+}
+
+interface RangeSliderProperties {
+    data: FinanceOperation[];
+    onChange?: (filteredData: FinanceOperation[]) => void;
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/range-slider/range-slider.tsx:14]:
```suggestion
    onChange?: (filteredData: {amount: number}[]) => void;
```

File: frontend/src/bundles/common/components/range-slider/range-slider.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import 'rc-slider/assets/index.css';
+
+import Slider from 'rc-slider';
+import React, { useCallback, useState } from 'react';
+
+import styles from './range-slider.module.scss';
+
+interface FinanceOperation {
+    amount: number;
+}
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/range-slider/range-slider.tsx:10]:
```suggestion
```

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]
You should add this component to `/ui` page

File: frontend/src/bundles/common/components/range-slider/range-slider.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+interface RangeSliderProperties {
+    onChange?: (range: { min: number; max: number }) => void;
+    rangeLimits?: { min: number; max: number };
+}
+
+const RangeSlider: React.FC<RangeSliderProperties> = ({
+    onChange,
+    rangeLimits = { min: -100, max: 1000 },
+}) => {
+    const [range, setRange] = useState([rangeLimits.min, rangeLimits.max]);
```
Comment (VItaliy-Kalenichenko) [inline frontend/src/bundles/common/components/range-slider/range-slider.tsx:18]: I think this state is redundant. You will have the same in the parent component.
So you have to add another prop, named like currentRange

File: frontend/src/bundles/common/components/range-slider/range-slider.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import '../../../../assets/css/variables/color-variables.css';
+
+import Slider from 'rc-slider';
+import React, { useCallback, useState } from 'react';
+
+import styles from './styles.module.scss';
+
+interface RangeSliderProperties {
+    onChange?: (range: { min: number; max: number }) => void;
+    rangeLimits?: { min: number; max: number };
```
Comment (VItaliy-Kalenichenko) [inline frontend/src/bundles/common/components/range-slider/range-slider.tsx:11]: Move rangeLimits, and onChange args types to a separate interface

File: frontend/src/bundles/common/components/range-slider/range-slider.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        },
+        [onChange],
+    );
+
+    const trackStyle = {
+        backgroundColor: 'var(--color-pink-100)',
+    };
+
+    const handleStyle = {
+        borderColor: 'var(--color-pink-100)',
```
Comment (VItaliy-Kalenichenko) [inline frontend/src/bundles/common/components/range-slider/range-slider.tsx:36]: I would rather create one 'styles' object with trackStyle and handleStyle. But I don't insist, so it's up to you

**Comment by MstyslavDmytryk** [comment]
slider

File: frontend/src/bundles/common/components/range-slider/range-slider.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
+import 'rc-slider/assets/index.css';
+import '../../../../assets/css/variables/color-variables.css';
+
+import Slider from 'rc-slider';
+import React, { useCallback } from 'react';
+
+import styles from './styles.module.scss';
+
+interface RangeLimits {
```
[the remark this answers is not in the document]
Comment (nikita-remeslov) [reply frontend/src/bundles/common/components/range-slider/range-slider.tsx:9]: Not resolved

File: frontend/src/bundles/common/components/range-slider/range-slider.tsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    min: number;
+    max: number;
+}
+
+interface RangeSliderProperties {
```
[the remark this answers is not in the document]
Comment (nikita-remeslov) [reply frontend/src/bundles/common/components/range-slider/range-slider.tsx:14]: Not resolved

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

**Review by anton-otroshchenko — APPROVED** [review]

**Review by VItaliy-Kalenichenko — APPROVED** [review]

**Pull request merged**